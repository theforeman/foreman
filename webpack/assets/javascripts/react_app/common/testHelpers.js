import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

export default {
  mockStorage: () => {
    const storage = {};

    return {
      setItem: (key, value) => {
        storage[key] = value || '';
      },
      getItem: key => (key in storage ? storage[key] : null),
      removeItem: key => {
        delete storage[key];
      },
      get length() {
        return Object.keys(storage).length;
      },
      key: i => {
        const keys = Object.keys(storage);

        return keys[i] || null;
      },
    };
  },
};

// a helper method for invoking a class method (for unit tests)
// obj = a class
// func = a tested function
// objThis = an object's this
// arg = function args

export const classFunctionUnitTest = (obj, func, objThis, args) =>
  obj.prototype[func].apply(objThis, args);

/**
 * Shallow render a component multipile times with fixtures
 * @param  {ReactComponent} Component Component to shallow-render
 * @param  {Object}         fixtures  key=fixture description, value=props to apply
 * @return {Object}                   key=fixture description, value=shallow-rendered component
 */
export const shallowRenderComponentWithFixtures = (Component, fixtures) =>
  Object.entries(fixtures).map(([description, props]) => ({
    description,
    component: shallow(<Component {...props} />),
  }));

/**
 * Test a component with fixtures and snapshots
 * @param  {ReactComponent} Component Component to test
 * @param  {Object}         fixtures  key=fixture description, value=props to apply
 */
export const testComponentSnapshotsWithFixtures = (Component, fixtures) =>
  shallowRenderComponentWithFixtures(Component, fixtures).forEach(
    ({ description, component }) =>
      it(description, () => expect(toJson(component)).toMatchSnapshot())
  );

/**
 * run an action (sync or async) and except the results to much snapshot
 * @param  {Function}  runAction  Action runner function
 * @return {Promise}
 */
export const testActionSnapshot = async runAction => {
  const actionResults = runAction();

  // if it's an async action
  if (typeof actionResults === 'function') {
    const dispatch = jest.fn();
    await actionResults(dispatch);

    expect(dispatch.mock.calls).toMatchSnapshot();
  } else {
    expect(actionResults).toMatchSnapshot();
  }
};

/**
 * Test actions with fixtures and snapshots
 * @param  {Object} fixtures key=fixture description, value=action runner function
 */
export const testActionSnapshotWithFixtures = fixtures =>
  Object.entries(fixtures).forEach(([description, runAction]) =>
    it(description, () => testActionSnapshot(runAction))
  );

/**
 * Test a reducer with fixtures and snapshots
 * @param  {Function} reducer  reducer to test
 * @param  {Object}   fixtures key=fixture description, value=props to apply
 */
export const testReducerSnapshotWithFixtures = (reducer, fixtures) => {
  const reduce = ({ state, action = {} } = {}) => reducer(state, action);
  Object.entries(fixtures).forEach(([description, action]) =>
    it(description, () => expect(reduce(action)).toMatchSnapshot())
  );
};

/**
 * Test selectors with fixtures and snapshots
 * @param  {Object} fixtures  key=fixture description,
 *                            value=selector runner function
 */
export const testSelectorsSnapshotWithFixtures = fixtures =>
  Object.entries(fixtures).forEach(([description, selectorRunner]) =>
    it(description, () => expect(selectorRunner()).toMatchSnapshot())
  );
