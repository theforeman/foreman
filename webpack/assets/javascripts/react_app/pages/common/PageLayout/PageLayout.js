import React from 'react';
import PropTypes from 'prop-types';
import { Row, Col } from 'patternfly-react';
import { resolveSearchQuery } from '../../../components/SearchBar/SearchBarHelpers';

import ToastsList from '../../../components/ToastsList';
import BreadcrumbBar from '../../../components/BreadcrumbBar';
import SearchBar from '../../../components/SearchBar';

const PageLayout = ({
  header,
  searchable,
  searchProps,
  searchQuery,
  onSearch,
  customBreadcrumbs,
  breadcrumbOptions,
  toolbarButtons,
  toastNotifications,
  children,
}) => (
  <div id="main">
    <div id="react-content">
      {toastNotifications && (
        <div
          id="toast-notifications-container"
          data-notifications={toastNotifications}
        >
          <ToastsList />
        </div>
      )}
      <div id="breadcrumb">
        {!breadcrumbOptions && (
          <div className="row form-group">
            <h1 className="col-md-8">{header}</h1>
          </div>
        )}
        {customBreadcrumbs
          ? { customBreadcrumbs }
          : breadcrumbOptions && <BreadcrumbBar data={breadcrumbOptions} />}
      </div>
      <Row>
        <Col className="title_filter" md={searchable ? 6 : 4}>
          {searchable && (
            <div id="search-bar">
              {
                <SearchBar
                  data={searchProps}
                  initialQuery={searchQuery}
                  onSearch={onSearch}
                />
              }
            </div>
          )}
        </Col>
        <Col id="title_action" md={searchable ? 6 : 8}>
          <div className="btn-toolbar pull-right">{toolbarButtons}</div>
        </Col>
      </Row>
      {children}
    </div>
  </div>
);

PageLayout.propTypes = {
  children: PropTypes.node.isRequired,
  searchable: PropTypes.bool.isRequired,
  header: PropTypes.string,
  searchProps: PropTypes.shape({
    autocomplete: PropTypes.shape({
      results: PropTypes.array,
      searchQuery: PropTypes.string,
      url: PropTypes.string,
      useKeyShortcuts: PropTypes.bool,
    }),
    controller: PropTypes.string,
    bookmarks: PropTypes.shape({
      text: PropTypes.string,
      query: PropTypes.string,
    }),
  }),
  customBreadcrumbs: PropTypes.node,
  breadcrumbOptions: PropTypes.shape({
    isSwitchable: PropTypes.bool,
    resource: PropTypes.shape({
      nameField: PropTypes.string,
      resourceUrl: PropTypes.string,
      switcherItemUrl: PropTypes.string,
      resourceFilter: PropTypes.string,
    }),
    breadcrumbItems: PropTypes.arrayOf(
      PropTypes.shape({
        caption: PropTypes.oneOfType([
          PropTypes.string.isRequired,
          PropTypes.shape({
            icon: PropTypes.shape({
              url: PropTypes.string,
              alt: PropTypes.string,
            }),
            text: PropTypes.string,
          }),
        ]),
        url: PropTypes.string,
      })
    ),
  }),
  toolbarButtons: PropTypes.node,
  toastNotifications: PropTypes.string,
  onSearch: PropTypes.func,
  searchQuery: PropTypes.string,
};

PageLayout.defaultProps = {
  searchProps: {},
  header: '',
  searchQuery: '',
  toastNotifications: null,
  customBreadcrumbs: null,
  toolbarButtons: null,
  breadcrumbOptions: null,
  onSearch: resolveSearchQuery,
};

export default PageLayout;
