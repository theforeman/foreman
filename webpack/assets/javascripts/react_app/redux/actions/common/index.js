import API from '../../../API';

export const ajaxRequestAction = ({
  dispatch,
  requestAction,
  successAction,
  failedAction,
  url,
  item,
}) => {
  dispatch({ type: requestAction, payload: item });
  return API.get(url)
    .then(({ data }) =>
      dispatch({ type: successAction, payload: { ...item, ...data } })
    )
    .catch(error => dispatch({ type: failedAction, payload: { error, item } }));
};
