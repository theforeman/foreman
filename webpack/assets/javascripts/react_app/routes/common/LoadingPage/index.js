import React from 'react';
import { Spinner } from 'patternfly-react';
import './loadingpage.scss';

const LoadingPage = () => (
  <div id="loading-page">
    <Spinner loading size="lg" />
  </div>
);

export default LoadingPage;
