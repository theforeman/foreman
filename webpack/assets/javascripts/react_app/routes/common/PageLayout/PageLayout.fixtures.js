import { breadcrumbBar } from '../../../components/BreadcrumbBar/BreadcrumbBar.fixtures';
import { SearchBarProps } from '../../../components/SearchBar/SearchBar.fixtures';

export const pageLayoutMock = {
  header: 'Page',
  searchable: true,
  searchProps: SearchBarProps,
  customBreadcrumbs: null,
  breadcrumbOptions: breadcrumbBar,
  toolbarButtons: null,
  toastNotifications: null,
  children: 'body',
};
