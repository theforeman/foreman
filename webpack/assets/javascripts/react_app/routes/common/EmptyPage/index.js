import React from 'react';
import PropTypes from 'prop-types';
import { translate as __ } from '../../../common/I18n';
import DefaultEmptyState from '../../../components/common/EmptyState';
import './emptypage.scss';

const EmptyPage = ({ message: { type, text } }) => (
  <DefaultEmptyState
    icon={type === 'error' ? 'error-circle-o' : 'add-circle-o'}
    header={type === 'error' ? __('Error') : __('No Results')}
    description={text}
  />
);

EmptyPage.propTypes = {
  message: PropTypes.shape({
    type: PropTypes.oneOf(['empty', 'error']),
    text: PropTypes.string,
  }),
};

EmptyPage.defaultProps = {
  message: PropTypes.shape({
    type: 'empty',
    text: 'No Results',
  }),
};

export default EmptyPage;
