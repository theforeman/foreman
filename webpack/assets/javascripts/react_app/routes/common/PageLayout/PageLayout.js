import React from 'react';
import PropTypes from 'prop-types';
import { Row, Col, Spinner } from 'patternfly-react';
import { updateDocumentTitle } from '../../../common/document';
import { changeQuery } from '../../../common/urlHelpers';

import ToastsList from '../../../components/ToastsList';
import BreadcrumbBar from '../../../components/BreadcrumbBar';
import SearchBar from '../../../components/SearchBar';

const PageLayout = ({
  header,
  searchable,
  searchProps,
  searchQuery,
  onSearch,
  onBookmarkClick,
  customBreadcrumbs,
  breadcrumbOptions,
  toolbarButtons,
  toastNotifications,
  beforeToolbarComponent,
  isLoading,
  children,
}) => {
  updateDocumentTitle(header);
  return (
    <div id="main">
      <div id="react-content">
        <ToastsList railsMessages={toastNotifications} />
        <div id="breadcrumb">
          {!breadcrumbOptions && (
            <div className="row form-group">
              <h1 className="col-md-8">{header}</h1>
            </div>
          )}
          {customBreadcrumbs
            ? { customBreadcrumbs }
            : breadcrumbOptions && <BreadcrumbBar data={breadcrumbOptions} />}
        </div>
        {beforeToolbarComponent}
        <Row>
          <Col className="title_filter" md={searchable ? 6 : 4}>
            {searchable && (
              <SearchBar
                data={searchProps}
                initialQuery={searchQuery}
                onSearch={onSearch}
                onBookmarkClick={onBookmarkClick}
              />
            )}
            &nbsp;
          </Col>
          <Col id="title_action" md={searchable ? 6 : 8}>
            <div className="btn-toolbar pull-right">
              {isLoading && (
                <div id="toolbar-spinner">
                  <Spinner loading size="sm" />
                </div>
              )}
              {toolbarButtons}
            </div>
          </Col>
        </Row>
        {children}
      </div>
    </div>
  );
};

PageLayout.propTypes = {
  children: PropTypes.node.isRequired,
  searchable: PropTypes.bool.isRequired,
  header: PropTypes.string,
  searchProps: PropTypes.shape({
    autocomplete: PropTypes.shape({
      results: PropTypes.array,
      searchQuery: PropTypes.string,
      url: PropTypes.string,
      useKeyShortcuts: PropTypes.bool,
    }),
    controller: PropTypes.string,
    bookmarks: PropTypes.shape({
      text: PropTypes.string,
      query: PropTypes.string,
    }),
  }),
  customBreadcrumbs: PropTypes.node,
  breadcrumbOptions: PropTypes.shape({
    isSwitchable: PropTypes.bool,
    resource: PropTypes.shape({
      nameField: PropTypes.string,
      resourceUrl: PropTypes.string,
      switcherItemUrl: PropTypes.string,
      resourceFilter: PropTypes.string,
    }),
    breadcrumbItems: PropTypes.arrayOf(
      PropTypes.shape({
        caption: PropTypes.oneOfType([
          PropTypes.string.isRequired,
          PropTypes.shape({
            icon: PropTypes.shape({
              url: PropTypes.string,
              alt: PropTypes.string,
            }),
            text: PropTypes.string,
          }),
        ]),
        url: PropTypes.string,
      })
    ),
  }),
  toolbarButtons: PropTypes.node,
  toastNotifications: PropTypes.array,
  onSearch: PropTypes.func,
  onBookmarkClick: PropTypes.func,
  searchQuery: PropTypes.string,
  beforeToolbarComponent: PropTypes.node,
  isLoading: PropTypes.bool,
};

PageLayout.defaultProps = {
  searchProps: {},
  header: '',
  searchQuery: '',
  toastNotifications: [],
  customBreadcrumbs: null,
  toolbarButtons: null,
  breadcrumbOptions: null,
  isLoading: false,
  onSearch: searchQuery => changeQuery({ search: searchQuery.trim(), page: 1 }),
  onBookmarkClick: searchQuery =>
    changeQuery({ search: searchQuery.trim(), page: 1 }),
  beforeToolbarComponent: null,
};

export default PageLayout;
