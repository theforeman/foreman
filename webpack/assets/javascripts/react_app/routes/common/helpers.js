import { foremanUrl, urlBuilder } from '../../common/urlHelpers';

export const getControllerSearchProps = (
  controller,
  id = 'searchBar',
  canCreate = true
) => ({
  controller,
  autocomplete: {
    id,
    searchQuery: '',
    url: urlBuilder(controller, 'auto_complete_search'),
    useKeyShortcuts: true,
  },
  bookmarks: {
    url: foremanUrl('/api/bookmarks'),
    canCreate,
    documentationUrl: `4.1.5Searching`,
  },
});
