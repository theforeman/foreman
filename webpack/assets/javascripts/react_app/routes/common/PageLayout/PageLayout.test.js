import { testComponentSnapshotsWithFixtures } from '../../../common/testHelpers';
import PageLayout from './PageLayout';
import { pageLayoutMock } from './PageLayout.fixtures';

const pageLayoutFixtures = {
  'render pageLayout w/search': pageLayoutMock,
  'render pageLayout without search': { ...pageLayoutMock, searchable: false },
  'render pageLayout with custom breadcrumbs': {
    ...pageLayoutMock,
    customBreadcrumbs: 'customBreadcrumbs',
  },
  'render pageLayout without breadcrumbs': {
    ...pageLayoutMock,
    breadcrumbOptions: null,
  },
  'render pageLayout w/toastNotifications': {
    ...pageLayoutMock,
    toastNotifications: 'notification',
  },
  'render pageLayout w/toolBar': {
    ...pageLayoutMock,
    toolbarButtons: 'toolbarButton',
  },
  'render pageLayout w/beforeToolbarComponent': {
    ...pageLayoutMock,
    beforeToolbarComponent: 'beforeToolbarComponent',
  },
};

testComponentSnapshotsWithFixtures(PageLayout, pageLayoutFixtures);
