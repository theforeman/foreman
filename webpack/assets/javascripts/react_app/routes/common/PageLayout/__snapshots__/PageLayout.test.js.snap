// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render pageLayout w/beforeToolbarComponent 1`] = `
<div
  id="main"
>
  <div
    id="react-content"
  >
    <div
      id="breadcrumb"
    >
      <Connect(BreadcrumbBar)
        data={
          Object {
            "data": Object {
              "breadcrumbItems": Array [
                Object {
                  "caption": "root",
                  "url": "/some-url",
                },
                Object {
                  "caption": "child with onClick",
                  "onClick": [MockFunction],
                },
                Object {
                  "caption": "active child",
                },
              ],
              "isSwitchable": false,
              "resource": Object {
                "nameField": "name",
                "resourceUrl": "some/url",
                "switcherItemUrl": "some/url/:id",
              },
            },
          }
        }
      />
    </div>
    beforeToolbarComponent
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className="title_filter"
        componentClass="div"
        md={6}
      >
        <div
          id="search-bar"
        >
          <Connect(SearchBar)
            data={
              Object {
                "data": Object {
                  "autocomplete": Object {
                    "id": "some-id",
                    "searchQuery": null,
                    "url": "model/auto_complete_search",
                    "useKeyShortcuts": true,
                  },
                  "bookmarks": Object {
                    "canCreate": true,
                    "documentationUrl": "/doc/url",
                    "url": "/api/bookmarks",
                  },
                  "controller": "models",
                },
              }
            }
            initialQuery=""
            onBookmarkClick={[Function]}
            onSearch={[Function]}
          />
        </div>
      </Col>
      <Col
        bsClass="col"
        componentClass="div"
        id="title_action"
        md={6}
      >
        <div
          className="btn-toolbar pull-right"
        />
      </Col>
    </Row>
    body
  </div>
</div>
`;

exports[`render pageLayout w/search 1`] = `
<div
  id="main"
>
  <div
    id="react-content"
  >
    <div
      id="breadcrumb"
    >
      <Connect(BreadcrumbBar)
        data={
          Object {
            "data": Object {
              "breadcrumbItems": Array [
                Object {
                  "caption": "root",
                  "url": "/some-url",
                },
                Object {
                  "caption": "child with onClick",
                  "onClick": [MockFunction],
                },
                Object {
                  "caption": "active child",
                },
              ],
              "isSwitchable": false,
              "resource": Object {
                "nameField": "name",
                "resourceUrl": "some/url",
                "switcherItemUrl": "some/url/:id",
              },
            },
          }
        }
      />
    </div>
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className="title_filter"
        componentClass="div"
        md={6}
      >
        <div
          id="search-bar"
        >
          <Connect(SearchBar)
            data={
              Object {
                "data": Object {
                  "autocomplete": Object {
                    "id": "some-id",
                    "searchQuery": null,
                    "url": "model/auto_complete_search",
                    "useKeyShortcuts": true,
                  },
                  "bookmarks": Object {
                    "canCreate": true,
                    "documentationUrl": "/doc/url",
                    "url": "/api/bookmarks",
                  },
                  "controller": "models",
                },
              }
            }
            initialQuery=""
            onBookmarkClick={[Function]}
            onSearch={[Function]}
          />
        </div>
      </Col>
      <Col
        bsClass="col"
        componentClass="div"
        id="title_action"
        md={6}
      >
        <div
          className="btn-toolbar pull-right"
        />
      </Col>
    </Row>
    body
  </div>
</div>
`;

exports[`render pageLayout w/toastNotifications 1`] = `
<div
  id="main"
>
  <div
    id="react-content"
  >
    <div
      data-notifications="notification"
      id="toast-notifications-container"
    >
      <Connect(ToastsList) />
    </div>
    <div
      id="breadcrumb"
    >
      <Connect(BreadcrumbBar)
        data={
          Object {
            "data": Object {
              "breadcrumbItems": Array [
                Object {
                  "caption": "root",
                  "url": "/some-url",
                },
                Object {
                  "caption": "child with onClick",
                  "onClick": [MockFunction],
                },
                Object {
                  "caption": "active child",
                },
              ],
              "isSwitchable": false,
              "resource": Object {
                "nameField": "name",
                "resourceUrl": "some/url",
                "switcherItemUrl": "some/url/:id",
              },
            },
          }
        }
      />
    </div>
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className="title_filter"
        componentClass="div"
        md={6}
      >
        <div
          id="search-bar"
        >
          <Connect(SearchBar)
            data={
              Object {
                "data": Object {
                  "autocomplete": Object {
                    "id": "some-id",
                    "searchQuery": null,
                    "url": "model/auto_complete_search",
                    "useKeyShortcuts": true,
                  },
                  "bookmarks": Object {
                    "canCreate": true,
                    "documentationUrl": "/doc/url",
                    "url": "/api/bookmarks",
                  },
                  "controller": "models",
                },
              }
            }
            initialQuery=""
            onBookmarkClick={[Function]}
            onSearch={[Function]}
          />
        </div>
      </Col>
      <Col
        bsClass="col"
        componentClass="div"
        id="title_action"
        md={6}
      >
        <div
          className="btn-toolbar pull-right"
        />
      </Col>
    </Row>
    body
  </div>
</div>
`;

exports[`render pageLayout w/toolBar 1`] = `
<div
  id="main"
>
  <div
    id="react-content"
  >
    <div
      id="breadcrumb"
    >
      <Connect(BreadcrumbBar)
        data={
          Object {
            "data": Object {
              "breadcrumbItems": Array [
                Object {
                  "caption": "root",
                  "url": "/some-url",
                },
                Object {
                  "caption": "child with onClick",
                  "onClick": [MockFunction],
                },
                Object {
                  "caption": "active child",
                },
              ],
              "isSwitchable": false,
              "resource": Object {
                "nameField": "name",
                "resourceUrl": "some/url",
                "switcherItemUrl": "some/url/:id",
              },
            },
          }
        }
      />
    </div>
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className="title_filter"
        componentClass="div"
        md={6}
      >
        <div
          id="search-bar"
        >
          <Connect(SearchBar)
            data={
              Object {
                "data": Object {
                  "autocomplete": Object {
                    "id": "some-id",
                    "searchQuery": null,
                    "url": "model/auto_complete_search",
                    "useKeyShortcuts": true,
                  },
                  "bookmarks": Object {
                    "canCreate": true,
                    "documentationUrl": "/doc/url",
                    "url": "/api/bookmarks",
                  },
                  "controller": "models",
                },
              }
            }
            initialQuery=""
            onBookmarkClick={[Function]}
            onSearch={[Function]}
          />
        </div>
      </Col>
      <Col
        bsClass="col"
        componentClass="div"
        id="title_action"
        md={6}
      >
        <div
          className="btn-toolbar pull-right"
        >
          toolbarButton
        </div>
      </Col>
    </Row>
    body
  </div>
</div>
`;

exports[`render pageLayout with custom breadcrumbs 1`] = `
<div
  id="main"
>
  <div
    id="react-content"
  >
    <div
      id="breadcrumb"
    >
      <Component />
    </div>
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className="title_filter"
        componentClass="div"
        md={6}
      >
        <div
          id="search-bar"
        >
          <Connect(SearchBar)
            data={
              Object {
                "data": Object {
                  "autocomplete": Object {
                    "id": "some-id",
                    "searchQuery": null,
                    "url": "model/auto_complete_search",
                    "useKeyShortcuts": true,
                  },
                  "bookmarks": Object {
                    "canCreate": true,
                    "documentationUrl": "/doc/url",
                    "url": "/api/bookmarks",
                  },
                  "controller": "models",
                },
              }
            }
            initialQuery=""
            onBookmarkClick={[Function]}
            onSearch={[Function]}
          />
        </div>
      </Col>
      <Col
        bsClass="col"
        componentClass="div"
        id="title_action"
        md={6}
      >
        <div
          className="btn-toolbar pull-right"
        />
      </Col>
    </Row>
    body
  </div>
</div>
`;

exports[`render pageLayout without breadcrumbs 1`] = `
<div
  id="main"
>
  <div
    id="react-content"
  >
    <div
      id="breadcrumb"
    >
      <div
        className="row form-group"
      >
        <h1
          className="col-md-8"
        >
          Page
        </h1>
      </div>
    </div>
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className="title_filter"
        componentClass="div"
        md={6}
      >
        <div
          id="search-bar"
        >
          <Connect(SearchBar)
            data={
              Object {
                "data": Object {
                  "autocomplete": Object {
                    "id": "some-id",
                    "searchQuery": null,
                    "url": "model/auto_complete_search",
                    "useKeyShortcuts": true,
                  },
                  "bookmarks": Object {
                    "canCreate": true,
                    "documentationUrl": "/doc/url",
                    "url": "/api/bookmarks",
                  },
                  "controller": "models",
                },
              }
            }
            initialQuery=""
            onBookmarkClick={[Function]}
            onSearch={[Function]}
          />
        </div>
      </Col>
      <Col
        bsClass="col"
        componentClass="div"
        id="title_action"
        md={6}
      >
        <div
          className="btn-toolbar pull-right"
        />
      </Col>
    </Row>
    body
  </div>
</div>
`;

exports[`render pageLayout without search 1`] = `
<div
  id="main"
>
  <div
    id="react-content"
  >
    <div
      id="breadcrumb"
    >
      <Connect(BreadcrumbBar)
        data={
          Object {
            "data": Object {
              "breadcrumbItems": Array [
                Object {
                  "caption": "root",
                  "url": "/some-url",
                },
                Object {
                  "caption": "child with onClick",
                  "onClick": [MockFunction],
                },
                Object {
                  "caption": "active child",
                },
              ],
              "isSwitchable": false,
              "resource": Object {
                "nameField": "name",
                "resourceUrl": "some/url",
                "switcherItemUrl": "some/url/:id",
              },
            },
          }
        }
      />
    </div>
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        className="title_filter"
        componentClass="div"
        md={4}
      />
      <Col
        bsClass="col"
        componentClass="div"
        id="title_action"
        md={8}
      >
        <div
          className="btn-toolbar pull-right"
        />
      </Col>
    </Row>
    body
  </div>
</div>
`;
